js/sliderView.js

```diff
@@ -289,74 +291,77 @@ define([
+            const topOfRange = this.model.get('_correctRange')._top;
+            while (answer <= topOfRange) {
+              answers.push(answer);
+              answer += step;
+            }
+          }
+        } else {
+          console.log('adapt-contrib-slider::WARNING: no correct answer or correct range set in JSON');
         }
-      } else {
-        console.log('adapt-contrib-slider::WARNING: no correct answer or correct range set in JSON');
       }
 
-      var middleAnswer = answers[Math.floor(answers.length / 2)];
+      const middleAnswer = answers[Math.floor(answers.length / 2)];
       this.animateToPosition(this.mapIndexToPixels(this.getIndexFromValue(middleAnswer)));
 
       this.showModelAnswers(answers);
 
       this.setSliderValue(middleAnswer);
-    },
+    }
 
-    showModelAnswers: function(correctAnswerArray) {
-      var $parentDiv = this.$('.js-slider-model-range');
-      _.each(correctAnswerArray, function(correctAnswer, index) {
+    showModelAnswers(correctAnswerArray) {
+      const $parentDiv = this.$('.js-slider-model-range');
+      correctAnswerArray.forEach((correctAnswer, index) => {
         $parentDiv.append($('<div class="slider__number-model-answer">'));
 
-        var $element = $(this.$('.js-slider-model-range .slider__number-model-answer')[index]);
-        var startingLeft = this.mapIndexToPixels(this.getIndexFromValue(this.model.get('_selectedItem').value));
+        const $element = $(this.$('.js-slider-model-range .slider__number-model-answer')[index]);
+        const startingLeft = this.mapIndexToPixels(this.getIndexFromValue(this.model.get('_selectedItem').value));
 
         if (this.model.get('_showNumber')) $element.html(correctAnswer);
 
-        $element.css({left:startingLeft}).fadeIn(0, _.bind(function() {
-          $element.animate({left: this.mapIndexToPixels(this.getIndexFromValue(correctAnswer))});
-        }, this));
-      }, this);
-    },
+        $element.css({ left: startingLeft }).fadeIn(0, () => {
+          $element.animate({ left: this.mapIndexToPixels(this.getIndexFromValue(correctAnswer)) });
+        });
+      });
+    }
 
-    hideCorrectAnswer: function() {
-      var userAnswerIndex = this.getIndexFromValue(this.model.get('_userAnswer'));
+    hideCorrectAnswer() {
+      const userAnswerIndex = this.getIndexFromValue(this.model.get('_userAnswer'));
       this.$('.js-slider-model-range').empty();
 
       this.showScaleMarker(true);
-      this.selectItem(userAnswerIndex, true);
+      this.selectItem(userAnswerIndex);
       this.animateToPosition(this.mapIndexToPixels(userAnswerIndex));
       this.setSliderValue(this.model.get('_userAnswer'));
-    },
+    }
 
     // according to given item index this should make the item as selected
-    selectItem: function(itemIndex, noFocus) {
-      _.each(this.model.get('_items'), function(item, index) {
+    selectItem(itemIndex) {
+      this.model.get('_items').forEach((item, index) => {
```

Comment:
Could refactor to remove the loop:
```js
const item = this.model.get('_items')[itemIndex];
if (!item) return;
item.selected = true;
this.model.set('_selectedItem', item);
this.$('.js-slider-item-input').attr({
  'value': item.value,
  'aria-valuenow': item.value
});
```

Reply:
I have resolved all of your comments. Just a note on this, the first item is defaulted to be selected so have called on deselectAllItems before I select the chosen item to be sure there is not anything currently selected. If we dont do this the value of isCorrect could be wrong, so just to be sure I want to deselect them all and not just the first item.